src/Contacts/Contact.m

```diff
@@ -67,45 +65,58 @@ - (instancetype)initWithSystemContact:(CNContact *)contact
+                phoneNumberNameMap[phoneNumber.stringValue]
+                    = NSLocalizedString(@"PHONE_NUMBER_TYPE_MAIN", @"Label for 'Main' phone numbers.");
             } else if ([phoneNumberField.label isEqualToString:CNLabelPhoneNumberHomeFax]) {
-                phoneNumberTypeMap[phoneNumber.stringValue] = @(OWSPhoneNumberTypeHomeFAX);
+                phoneNumberNameMap[phoneNumber.stringValue]
+                    = NSLocalizedString(@"PHONE_NUMBER_TYPE_HOME_FAX", @"Label for 'HomeFAX' phone numbers.");
             } else if ([phoneNumberField.label isEqualToString:CNLabelPhoneNumberWorkFax]) {
-                phoneNumberTypeMap[phoneNumber.stringValue] = @(OWSPhoneNumberTypeWorkFAX);
+                phoneNumberNameMap[phoneNumber.stringValue]
+                    = NSLocalizedString(@"PHONE_NUMBER_TYPE_WORK_FAX", @"Label for 'Work FAX' phone numbers.");
             } else if ([phoneNumberField.label isEqualToString:CNLabelPhoneNumberOtherFax]) {
-                phoneNumberTypeMap[phoneNumber.stringValue] = @(OWSPhoneNumberTypeOtherFAX);
+                phoneNumberNameMap[phoneNumber.stringValue]
+                    = NSLocalizedString(@"PHONE_NUMBER_TYPE_OTHER_FAX", @"Label for 'Other FAX' phone numbers.");
             } else if ([phoneNumberField.label isEqualToString:CNLabelPhoneNumberPager]) {
-                phoneNumberTypeMap[phoneNumber.stringValue] = @(OWSPhoneNumberTypePager);
+                phoneNumberNameMap[phoneNumber.stringValue]
+                    = NSLocalizedString(@"PHONE_NUMBER_TYPE_PAGER", @"Label for 'Pager' phone numbers.");
             } else if ([phoneNumberField.label isEqualToString:CNLabelOther]) {
-                phoneNumberTypeMap[phoneNumber.stringValue] = @(OWSPhoneNumberTypeOther);
-            } else {
-                // TODO this should only be hit if the user has a custom label, which ideally we'd expose rather than
-                // unknown. I think the thing to do is to get rid of OWSPhoneNumberType*, and simply localize the label
-                // in this method - uising the custom label when available.
-                // Maybe that means adding a "label" to PhoneNumber class and parsing the numbers in this class.
-                phoneNumberTypeMap[phoneNumber.stringValue] = @(OWSPhoneNumberTypeUnknown);
+                phoneNumberNameMap[phoneNumber.stringValue]
+                    = NSLocalizedString(@"PHONE_NUMBER_TYPE_OTHER", @"Label for 'Other' phone numbers.");
+            } else if (phoneNumberField.label.length > 0 && ![phoneNumberField.label hasPrefix:@"_$"]) {
+                // We'll reach this case for:
+                //
+                // * User-defined custom labels, which we want to display.
+                // * Labels like "_$!<CompanyMain>!$_", which I'm guessing are synced from other platforms.
+                //   We don't want to display these labels. Even some of iOS' default labels (like Radio) show
+                //   up this way.
```

Comment:
We could parse these with a regular expression, but I feel like this is an edge case we're already handling pretty well. I'm wary of relying on the details of how Apple handles these other labels.